### ISSUE TITLE
[Docs] Flytekit-Java Docs

### ISSUE THREAD

**Issue by brucearctor** [flyteorg/flyte#1254]
**Describe the documentation update**
Flytekit-java docs don't exist nicely like flytekit and other parts of flyte.  The link just goes to the github repo.  

It is likely early, but this would be to track what to use to generate docs, and highlight this is an [eventual ... perhaps not the priority, yet] nice-to-have.

<img width="1400" alt="Screen Shot 2021-07-16 at 1 45 04 PM" src="https://user-images.githubusercontent.com/5032356/126007123-e898096f-0dfc-4a8c-9be2-b2cd91b6427d.png">

**Comment by Smartmind12** [comment]
@brucearctor Can you assign this issue to me as i would like to work on it...Thanks!

**Comment by brucearctor** [comment]
@Smartmind12 - this issue isn't narrow enough, and is more of a meta-issue.  

There is related work [ in more tractable pieces ] to address.

Ex: 
* Generate Java Docs 
* Extend Java Docs in code [ for what gets generated ]
* Update webpage to point to Javadocs
* ... etc ...

Suggesting to create issues for each more tractable piece, and then address them one by one.  I'm happy to comment on things, if you link to it here and comment/tag me so that I am alerted.

**Comment by Smartmind12** [comment]
> @Smartmind12 - this issue isn't narrow enough, and is more of a meta-issue.
> 
> There is related work [ in more tractable pieces ] to address.
> 
> Ex:
> 
> * Generate Java Docs
> * Extend Java Docs in code [ for what gets generated ]
> * Update webpage to point to Javadocs
> * ... etc ...
> 
> Suggesting to create issues for each more tractable piece, and then address them one by one. I'm happy to comment on things, if you link to it here and comment/tag me so that I am alerted.

It would be helpful if you could create a list of all subsections you want in the document??...So considering those i will move forward and implement

**Comment by kumare3** [comment]
cc @honnix what is the plan for getting docs for java sdk?

**Comment by honnix** [comment]
If this is about javadoc, they are already generated and published to Maven central. (The quality is a not top but that we can improve.) However if this is about generating a Maven site which includes browsable javadoc, we don't have one yet and we will need to figure out where to host it. Maybe github pages.

**Comment by honnix** [comment]
Maybe we can use https://javadoc.io/ to just host javadocs. That seems to be easier.

**Comment by kumare3** [comment]
Is this the right link - https://javadoc.io/doc/org.flyte/flytekit-java/latest/index.html.
@honnix would love a getting started that works like pyflyte run

**Comment by honnix** [comment]
Yes that link is correct but I'm not exactly sure it is OK to use that service. I will ask around in my org.

I think there is still a long way to have a `pyflyte run` equivalent, because jflyte doesn't do any runtime stuff and there is also generally a lack of oauth module in jflyte. I will bring this back to the team.

**Comment by kumare3** [comment]
I don't think knitting has to be Flyte run, but package and flytectl register is perfectly fine as a jump of point. I think the audiences are different so this should be ok

**Comment by honnix** [comment]
That is a very reasonable feature to have. cc @andresgomezfrr @pablocasares

**Comment by honnix** [comment]
I think it is a matter of documentation then because jflyte already supports writing proto files to a folder. tar is not support but flytectl can handle individual files well.
